src/Clash/Cores/Arp.hs

```diff
@@ -0,0 +1,72 @@
+{-|
+Module      : Clash.Cores.Arp
+Description : Provides a fully functional ARP stack.
+-}
+
+{-# language FlexibleContexts #-}
+{-# OPTIONS -fplugin=Protocols.Plugin #-}
+
+module Clash.Cores.Arp where
+
+import Clash.Prelude
+
+import Protocols
+import Protocols.Df qualified as Df
+
+import Clash.Cores.Arp.ArpManager
+import Clash.Cores.Arp.ArpReceiver
+import Clash.Cores.Arp.ArpTable
+import Clash.Cores.Arp.ArpTransmitter
+import Clash.Cores.Arp.ArpTypes
+import Clash.Cores.Ethernet.EthernetTypes
+import Clash.Cores.Ethernet.PacketStream
+import Clash.Cores.IP.IPv4Types
+
+
+-- | A fully functional ARP stack which handles ARP lookups from client circuits.
+--   Maintains a single-entry ARP table which the client circuit can query via the
+--   `ArpLookup` input. If the client-supplied IPv4 address is not found in the table,
+--   it transmits an ARP request for this specific address. The circuit will assert
+--   backpressure until either a reply has been received, or a timeout occurs. The
+--   maximum number of seconds the stack will wait for a reply to this request is
+--   configurable. The timeout (in seconds) of ARP table entries is configurable as well.
+--   All timeouts may be up to a second inaccurate.
+--
+--   Moreover, it takes in an Ethernet stream with the ARP
+--   etherType (0x0806), and updates the ARP table upon receiving a valid ARP
+--   reply or gratitious ARP request. Gratitious ARP replies are ignored for now.
+--   If a normal ARP request is received, it transmits a reply.
+--
+--   Does not support Proxy ARP.
+arpC
+  :: forall
+     (dom :: Domain)
+     (maxAgeSeconds :: Nat)
+     (maxWaitSeconds :: Nat)
```

Comment:
I think a wait period specified in seconds is to granular. Like a lot of times even just 1 second is too long to wait. Especially since our second timer is inaccurate upto a second. So you could specify wait time of 1 second and then get a timeout after 1ms if you are unlucky.

So we probably want to make a more granular counter, 1/16th of second maybe? And use that. But this is just something of note. Not blocking for me.